### DISCUSSION TITLE
Bean Temp curve trailing from CHARGE index

### DISCUSSION THREAD

**Discussion by JSTCOFFEE** [artisan-roaster-scope/artisan#942] [General]
I would like the life file to trail from wherever the CHARGE event is by a certain number of steps. I made a virtual device to do this but it doesn't trail from CHARGE. E.G. I've got Y2[-30] working, but I want that to be delayed from the CHARGE index and not the beginning of the log.

**Comment by MAKOMO** [comment]
Could you expand a bit on the problem you try to tackle? Your description does not give me a clear picture of what you want to achieve. Sorry.

**Reply by MAKOMO** [reply]
Index access to temperature data works only with fixed indices. A variable index like CHARGEd cannot be used in squared brackets. So Y2[dCHARGE-30] wont't work as symbolic formula. You might extend the function eval_math_expression() in main.py and run from source if you need this.

**Reply by JSTCOFFEE** [reply]
Due to restrictions on my probe placement by the roaster design, I'm trying to approximate the actual bean temperature to some degree by looking at the trailing curve as a rough model of that. I wanted to trail from CHARGE to keep things consistent. I'm not roasting off the trailing curve, just using it as another tool to make decisions. Does that help clear things up?
